In [1]:
# let's first install the selenium library
! pip install selenium

In [2]:
# Let's now import all the required libraries.
import selenium
import pandas as pd
from selenium import webdriver

In [3]:
# let's first connect to the web driver
driver = webdriver.Chrome(r"C:\Users\Neha\Downloads\chromedriver_win32\chromedriver.exe")

Q1: Write a python program to scrape data for “Data Analyst” Job position in 
“Bangalore” location. You have to scrape the job-title, job-location, company_name,
experience_required. You have to scrape first 10 jobs data

In [4]:
url = "https://www.naukri.com"
driver.get(url)

In [9]:
# finding element for job search bar
search_job = driver.find_element_by_xpath("//input[@class='sugInp']")
search_job

<selenium.webdriver.remote.webelement.WebElement (session="d8e71b3c3e3f839d9496870de33b48d6", element="8e07a531-9164-4a45-a7f2-8b67ad5b01fb")>

In [14]:
# write on search bar
search_job.send_keys('Data Analyst')

In [15]:
# finding element for job location bar
search_loc=driver.find_element_by_id('qsb-location-sugg')
search_loc.send_keys("Bangalore")

In [17]:
search_btn= driver .find_element_by_xpath("//button[@class='btn']")
search_btn

<selenium.webdriver.remote.webelement.WebElement (session="d8e71b3c3e3f839d9496870de33b48d6", element="05243454-1895-4054-8de1-33deddd15182")>

In [19]:
# clicking  using xpath function
search_btn=driver.find_element_by_xpath("//button[@class='btn']")
search_btn.click()

In [21]:
#so let's extract all the tags having the job titles
title_tags=driver.find_elements_by_xpath("//a[@class='title fw500 ellipsis']")
title_tags

[<selenium.webdriver.remote.webelement.WebElement (session="d8e71b3c3e3f839d9496870de33b48d6", element="48cbdbaa-13ff-4507-9849-98e7d166106b")>,
 <selenium.webdriver.remote.webelement.WebElement (session="d8e71b3c3e3f839d9496870de33b48d6", element="3a678c9e-5d8d-4b33-b1c2-8fa1e7d951b4")>,
 <selenium.webdriver.remote.webelement.WebElement (session="d8e71b3c3e3f839d9496870de33b48d6", element="30f2bec4-f50b-4b6e-b96c-a11008a3c3ae")>,
 <selenium.webdriver.remote.webelement.WebElement (session="d8e71b3c3e3f839d9496870de33b48d6", element="a74843c4-6e6f-4056-97ff-7086eafccb50")>,
 <selenium.webdriver.remote.webelement.WebElement (session="d8e71b3c3e3f839d9496870de33b48d6", element="dbd0125c-f3a1-4c91-bb93-ffd5baeb0e13")>,
 <selenium.webdriver.remote.webelement.WebElement (session="d8e71b3c3e3f839d9496870de33b48d6", element="8a8358b6-cb01-4054-a934-1bafa4d82d8d")>,
 <selenium.webdriver.remote.webelement.WebElement (session="d8e71b3c3e3f839d9496870de33b48d6", element="cc21611c-acce-444e-9754-0a

In [29]:
# extract the text of the job title  from the tags
job_titles=[]
for i in title_tags:
    if i.text is None:
        job_titles.append('Not')
    else:
        job_titles.append(i.text)
job_titles[:10]        

['Data Scientist / Data Analyst -Business Analyst',
 'Senior Data Analyst - Data Lineage',
 'Data analysts',
 'Data Analyst - Informatica MDM',
 'Excel VBA Jobs Bangalore | VBA data analyst Jobs',
 'Data analysts',
 'Business Data Analyst',
 'Data Analyst',
 'Data Analyst',
 'Data Analyst - Category Demand Management (Revenue & Growth)']

In [30]:
# lets extract all the tags having company names
company_tags=driver.find_elements_by_xpath("//a[@class='subTitle ellipsis fleft']")
company_tags

[<selenium.webdriver.remote.webelement.WebElement (session="d8e71b3c3e3f839d9496870de33b48d6", element="a1d8740a-cbf1-4314-b6e7-6896ac900646")>,
 <selenium.webdriver.remote.webelement.WebElement (session="d8e71b3c3e3f839d9496870de33b48d6", element="65e28939-50e4-4ec9-b8fa-5265da3c420e")>,
 <selenium.webdriver.remote.webelement.WebElement (session="d8e71b3c3e3f839d9496870de33b48d6", element="8a979aa7-cbf9-4e1b-b9d9-727e59ccd278")>,
 <selenium.webdriver.remote.webelement.WebElement (session="d8e71b3c3e3f839d9496870de33b48d6", element="d25f1168-f469-4681-9f03-8eb86a60a3fe")>,
 <selenium.webdriver.remote.webelement.WebElement (session="d8e71b3c3e3f839d9496870de33b48d6", element="bafe2bf5-7ce3-40f6-b972-8bcc9e86d2e1")>,
 <selenium.webdriver.remote.webelement.WebElement (session="d8e71b3c3e3f839d9496870de33b48d6", element="5f34f475-968f-4a65-9ac7-06cff0269fb3")>,
 <selenium.webdriver.remote.webelement.WebElement (session="d8e71b3c3e3f839d9496870de33b48d6", element="1a51d6ee-8364-46e2-bc6b-ad

In [33]:
# Now we will extract the text from the tags by looping over these tags
companies_names=[]

for i in company_tags:
    companies_names.append(i.text)
companies_names[:10]    

['Inflexion Analytix Private Limited',
 'Opex Global Services',
 'IBM India Pvt. Limited',
 'Shell India Markets Private Limited',
 'Mind Circus Innovation',
 'IBM India Pvt. Limited',
 'dotSolved India Pvt., Ltd.',
 'Myntra Designs Pvt. Ltd.',
 'Myntra Designs Pvt. Ltd.',
 'Myntra Designs Pvt. Ltd.']

In [36]:
# so lets extract  all the tags having the experience required data
experience_tags=driver.find_elements_by_xpath("//li[@class='fleft grey-text br2 placeHolderLi experience'] //span")
experience_tags

[<selenium.webdriver.remote.webelement.WebElement (session="d8e71b3c3e3f839d9496870de33b48d6", element="2bbe2ee1-19c8-4c4d-852d-51a19926206d")>,
 <selenium.webdriver.remote.webelement.WebElement (session="d8e71b3c3e3f839d9496870de33b48d6", element="852ef2ab-3703-4f39-863f-3b5d53210e2d")>,
 <selenium.webdriver.remote.webelement.WebElement (session="d8e71b3c3e3f839d9496870de33b48d6", element="8d9c2f03-a5ab-4faa-ae78-4f50b0ce210e")>,
 <selenium.webdriver.remote.webelement.WebElement (session="d8e71b3c3e3f839d9496870de33b48d6", element="ebc7043a-5958-45f1-bcc6-2edf4e6b5327")>,
 <selenium.webdriver.remote.webelement.WebElement (session="d8e71b3c3e3f839d9496870de33b48d6", element="3efc2fd4-9b73-4850-9f01-ae9b7d2970e2")>,
 <selenium.webdriver.remote.webelement.WebElement (session="d8e71b3c3e3f839d9496870de33b48d6", element="a4d754fd-560d-46e7-ad4f-2c82fb55b7f7")>,
 <selenium.webdriver.remote.webelement.WebElement (session="d8e71b3c3e3f839d9496870de33b48d6", element="539c185e-b4b9-438e-9f9f-ed

In [38]:
# no we will extract the text  from these tags only by one by looping over these tags
experience_list=[]
for i in experience_tags:
    experience_list.append(i.text)
experience_list[:10]   

['0-3 Yrs',
 '5-10 Yrs',
 '5-6 Yrs',
 '6-9 Yrs',
 '0-1 Yrs',
 '3-5 Yrs',
 '9-13 Yrs',
 '3-5 Yrs',
 '3-8 Yrs',
 '1-4 Yrs']

In [40]:
locations_tags=driver.find_elements_by_xpath("//li[@class='fleft grey-text br2 placeHolderLi location']/span")
locations_tags

[<selenium.webdriver.remote.webelement.WebElement (session="d8e71b3c3e3f839d9496870de33b48d6", element="6c663e32-a5e7-4699-8614-2dff752e0c5e")>,
 <selenium.webdriver.remote.webelement.WebElement (session="d8e71b3c3e3f839d9496870de33b48d6", element="97b19045-90d7-4fab-a36a-3731c7f3e22d")>,
 <selenium.webdriver.remote.webelement.WebElement (session="d8e71b3c3e3f839d9496870de33b48d6", element="6ba97d71-9582-4c1d-b961-1dfedb701f8f")>,
 <selenium.webdriver.remote.webelement.WebElement (session="d8e71b3c3e3f839d9496870de33b48d6", element="48691cf1-ab76-424d-9e72-2c8aaf319d8e")>,
 <selenium.webdriver.remote.webelement.WebElement (session="d8e71b3c3e3f839d9496870de33b48d6", element="cc32dae9-6c3e-42e4-8459-648d22460c71")>,
 <selenium.webdriver.remote.webelement.WebElement (session="d8e71b3c3e3f839d9496870de33b48d6", element="eff92e60-989b-4058-8b4f-1f47982ad3d6")>,
 <selenium.webdriver.remote.webelement.WebElement (session="d8e71b3c3e3f839d9496870de33b48d6", element="44e300f7-1112-4469-a474-fa

In [42]:
locations_list=[]
for i in locations_tags:
    locations_list.append(i.text)
locations_list[:10]    

['Mumbai, Hyderabad/Secunderabad, Pune, Gurgaon/Gurugram, Chennai, Bangalore/Bengaluru',
 'Bangalore/Bengaluru',
 'Bengaluru/Bangalore',
 'Bangalore/Bengaluru',
 'Mysore/Mysuru, Coimbatore, Bangalore/Bengaluru',
 'Bangalore/Bengaluru',
 'Chennai, Bangalore/Bengaluru, Mumbai (All Areas)',
 'Bangalore/Bengaluru',
 'Bangalore/Bengaluru',
 'Bangalore/Bengaluru']

In [43]:
#So lets check th length of ech element.
print(len(job_titles[:10])),print(len(companies_names[:10])),print(len(experience_list[:10])),print(len(locations_list[:10]))

10
10
10
10


(None, None, None, None)

# Creating a DataFarme for the Data Analyst jobs

In [47]:
jobs=pd.DataFrame({})
jobs['title']=job_titles[:10]
jobs['company']=companies_names[:10]
jobs['experience_required']=experience_list[:10]
jobs['location']=locations_list[:10]

In [48]:
jobs

title  \
0    Data Scientist / Data Analyst -Business Analyst   
1                 Senior Data Analyst - Data Lineage   
2                                      Data analysts   
3                     Data Analyst - Informatica MDM   
4   Excel VBA Jobs Bangalore | VBA data analyst Jobs   
5                                      Data analysts   
6                              Business Data Analyst   
7                                       Data Analyst   
8                                       Data Analyst   
9  Data Analyst - Category Demand Management (Rev...   

                               company experience_required  \
0   Inflexion Analytix Private Limited             0-3 Yrs   
1                 Opex Global Services            5-10 Yrs   
2               IBM India Pvt. Limited             5-6 Yrs   
3  Shell India Markets Private Limited             6-9 Yrs   
4               Mind Circus Innovation             0-1 Yrs   
5               IBM India Pvt. Limited             3-5 Yrs   
6           dotSolved India Pvt., Ltd.            9-13 Yrs   
7             Myntra Designs Pvt. Ltd.             3-5 Yrs   
8             Myntra Designs Pvt. Ltd.             3-8 Yrs   
9             Myntra Designs Pvt. Ltd.             1-4 Yrs   

                                            location  
0  Mumbai, Hyderabad/Secunderabad, Pune, Gurgaon/...  
1                                Bangalore/Bengaluru  
2                                Bengaluru/Bangalore  
3                                Bangalore/Bengaluru  
4     Mysore/Mysuru, Coimbatore, Bangalore/Bengaluru  
5                                Bangalore/Bengaluru  
6   Chennai, Bangalore/Bengaluru, Mumbai (All Areas)  
7                                Bangalore/Bengaluru  
8                                Bangalore/Bengaluru  
9                                Bangalore/Bengaluru

We can see that we have created the datset named jobs.

In [49]:
driver.close()

In [1]:
from selenium.common.exceptions import NoSuchElementException

# Ques 2

Write a python program to scrape data for “Data Scientist” Job position in “Bangalore” location. You have to scrape the job-title, job-location, company_name, full job-description. You have to scrape first 10 jobs data.

In [5]:
# let's first connect to the web driver
driver = webdriver.Chrome(r"C:\Users\Neha\Downloads\chromedriver_win32\chromedriver.exe")

In [6]:
url = "https://www.naukri.com"
driver.get(url)

In [7]:
# finding element for job search bar
search_job = driver.find_element_by_xpath("//input[@class='sugInp']")
search_job

<selenium.webdriver.remote.webelement.WebElement (session="6b4041c87deb129b933322334f4747af", element="fe5a71f3-bdf8-4850-bfac-fb7c61118aff")>

In [8]:
# write on search bar
search_job.send_keys('Data Scientist')

In [9]:
# finding element for job location bar
search_loc=driver.find_element_by_id('qsb-location-sugg')
search_loc.send_keys("Bangalore")

In [10]:
search_btn= driver .find_element_by_xpath("//button[@class='btn']")
search_btn

<selenium.webdriver.remote.webelement.WebElement (session="6b4041c87deb129b933322334f4747af", element="d45ff62e-2d26-4bb0-b98a-4016eacbb1f1")>

In [11]:
# clicking  using xpath function
search_btn=driver.find_element_by_class_name('btn')
search_btn.click()

In [12]:
#so let's extract all the tags having the job titles
title_tag=driver.find_elements_by_xpath("//a[@class='title fw500 ellipsis']")
title_tag

[<selenium.webdriver.remote.webelement.WebElement (session="6b4041c87deb129b933322334f4747af", element="e19191d2-b863-4d2c-8ad4-8585133ac604")>,
 <selenium.webdriver.remote.webelement.WebElement (session="6b4041c87deb129b933322334f4747af", element="67fcccfb-3ed8-43f0-8222-dc7d298b15f5")>,
 <selenium.webdriver.remote.webelement.WebElement (session="6b4041c87deb129b933322334f4747af", element="9f60df94-ca75-40c5-a338-13f6b33f7391")>,
 <selenium.webdriver.remote.webelement.WebElement (session="6b4041c87deb129b933322334f4747af", element="58501c22-ae51-436b-97d8-9d24e0790e91")>,
 <selenium.webdriver.remote.webelement.WebElement (session="6b4041c87deb129b933322334f4747af", element="b4705392-e5bd-4647-8531-5175c8f8889e")>,
 <selenium.webdriver.remote.webelement.WebElement (session="6b4041c87deb129b933322334f4747af", element="37ce5583-0288-40f1-a52f-623eebf8c227")>,
 <selenium.webdriver.remote.webelement.WebElement (session="6b4041c87deb129b933322334f4747af", element="3fc3173d-3a1c-4191-b7bd-44

In [13]:
# extract the text of the job title  from the tags
job1_titles=[]
for i in title_tag:
    if i.text is None:
        job1_titles.append('Not')
    else:
        job1_titles.append(i.text)
job1_titles[:10]        

['Data Scientist / Data Analyst -Business Analyst',
 'Senior Data Scientist',
 'Immediate job opening - Senior Data Scientist',
 'Senior Data Scientist',
 'VP - Sr. Data Scientist For Morgan Stanley, Bangalore',
 'Lead Data Scientist',
 'Lead Data Scientist',
 'Data analytics / Data scientist intern (work from Home)',
 'Data Scientist - IBM Garage',
 'Digital Enterprise Architect (Data Scientist)']

In [14]:
# lets extract all the tags having company names
company_tag=driver.find_elements_by_xpath("//a[@class='subTitle ellipsis fleft']")
company_tag

[<selenium.webdriver.remote.webelement.WebElement (session="6b4041c87deb129b933322334f4747af", element="4fde0edc-2502-4ae8-b9af-b63560adb8a9")>,
 <selenium.webdriver.remote.webelement.WebElement (session="6b4041c87deb129b933322334f4747af", element="0497b983-774c-4bd5-80e3-fffad6c2d227")>,
 <selenium.webdriver.remote.webelement.WebElement (session="6b4041c87deb129b933322334f4747af", element="7d8361f3-05bf-4487-96e6-6ab48234ce63")>,
 <selenium.webdriver.remote.webelement.WebElement (session="6b4041c87deb129b933322334f4747af", element="5338bd12-035e-4750-8f24-372695cbe997")>,
 <selenium.webdriver.remote.webelement.WebElement (session="6b4041c87deb129b933322334f4747af", element="7318cb0c-f856-4128-adef-4cc111918be5")>,
 <selenium.webdriver.remote.webelement.WebElement (session="6b4041c87deb129b933322334f4747af", element="6fd906cf-e509-469d-94c2-f85925fd8234")>,
 <selenium.webdriver.remote.webelement.WebElement (session="6b4041c87deb129b933322334f4747af", element="15b15fb6-58ff-4209-a69f-15

In [15]:
# Now we will extract the text from the tags by looping over these tags
companies1_names=[]

for i in company_tag:
    companies1_names.append(i.text)
companies1_names[:10]    

['Inflexion Analytix Private Limited',
 'RedBus',
 'RedBus',
 'Fractal Analytics',
 'Morgan Stanley Advantage Services',
 'FICO',
 'Fractal Analytics',
 'TalkValley LLC',
 'IBM India Pvt. Limited',
 'Mphasis Limited']

In [16]:
# lets extract all the tags having locations
locations_tag=driver.find_elements_by_xpath("//li[@class='fleft grey-text br2 placeHolderLi location']/span")
locations_tag

[<selenium.webdriver.remote.webelement.WebElement (session="6b4041c87deb129b933322334f4747af", element="5a06d9d0-4734-4d7c-b491-0209049d578d")>,
 <selenium.webdriver.remote.webelement.WebElement (session="6b4041c87deb129b933322334f4747af", element="c24fb833-fd4e-4416-b214-2e4b6b62669a")>,
 <selenium.webdriver.remote.webelement.WebElement (session="6b4041c87deb129b933322334f4747af", element="fe0bbb8f-88ea-471f-80d0-08151709f4b9")>,
 <selenium.webdriver.remote.webelement.WebElement (session="6b4041c87deb129b933322334f4747af", element="2a23be26-c2fb-44f0-b77c-f3c5d078398a")>,
 <selenium.webdriver.remote.webelement.WebElement (session="6b4041c87deb129b933322334f4747af", element="e8031cf7-9996-4776-b7c7-73674453e717")>,
 <selenium.webdriver.remote.webelement.WebElement (session="6b4041c87deb129b933322334f4747af", element="8aaa1e20-2b25-4760-acf9-19fe332b2b5b")>,
 <selenium.webdriver.remote.webelement.WebElement (session="6b4041c87deb129b933322334f4747af", element="8d2e0058-2142-4bf8-8bbc-61

In [17]:
locations1_list=[]
for i in locations_tag:
    locations1_list.append(i.text)
locations1_list[:10]    

['Mumbai, Hyderabad/Secunderabad, Pune, Gurgaon/Gurugram, Chennai, Bangalore/Bengaluru',
 'Bangalore/Bengaluru',
 'Bangalore/Bengaluru',
 'Mumbai, Gurgaon/Gurugram, Bangalore/Bengaluru',
 'Bangalore/Bengaluru',
 'Bangalore/Bengaluru',
 'Gurgaon/Gurugram, Bangalore/Bengaluru, Mumbai (All Areas)',
 'Kolkata, Bangalore/Bengaluru, Delhi / NCR',
 'Bengaluru/Bangalore',
 'Bangalore/Bengaluru']

In [18]:
print(len(job1_titles[:10])),print(len(companies1_names[:10])),print(len(locations1_list[:10]))

10
10
10


(None, None, None)

In [19]:
driver=webdriver.Chrome(r"C:\Users\Neha\Downloads\chromedriver_win32\chromedriver.exe")
driver.get('https://www.naukri.com/data-scientist-jobs-in-banglore-bagaluru')

In [20]:
urls=[]

In [21]:
job_description=[]

In [22]:
for i in driver.find_elements_by_xpath("//a[@class='title fw500 ellipsis']"):
    urls.append(i.get_attribute("href"))
        

In [23]:
for url in urls[:10]:
    
    
    try:
        driver.get(url)
        description=driver.find_element_by_xpath("//section[@class='job-desc']").text
        job_description.append(description)
        
    except NoSuchElementException:
        job_description.append("Not Available")

In [24]:
job_description

["Job description\nJob Role : Data Scientist/Data Analyst /Business Analyst\n\nLocation : Chennai/Bangalore/Hyderabad/Pune/Mumbai/Delhi\n\nGreetings from CAIA - Center for Artificial Intelligence & Advanced Analytics\n43% of companies experienced a high deficit of skilled resources with Advanced Analytical skills and AI implementing capabilities in year 2020. CAIA gives you a great opportunity to enter the world of future technologies and Innovations- Data Science, Analytics, AI, Data Visualization and Cloud Computing.\n\nWhile 2020 was a year like no other, we are living in an interesting times where data is reshaping the world, and businesses are rapidly adopting technology to gain an edge over others. Hence, there's a substantial increase in demand for technology professionals who can implement systems in data science, machine learning and AI in Tier 1 and Tier 2 organization's working closely with us.\n\nTo help you build a sustainable career we would like you to utilize data, soft

In [25]:
print(len(job_description))

10


# Creating a dataframe for the Data Analyst Jobs

In [27]:
job1=pd.DataFrame({})
job1['title']=job1_titles[:10]
job1['company_name']=companies1_names[:10]
job1['location']=locations1_list[:10]
job1['job_desc']=job_description

In [28]:
job1

title  \
0    Data Scientist / Data Analyst -Business Analyst   
1                              Senior Data Scientist   
2      Immediate job opening - Senior Data Scientist   
3                              Senior Data Scientist   
4  VP - Sr. Data Scientist For Morgan Stanley, Ba...   
5                                Lead Data Scientist   
6                                Lead Data Scientist   
7  Data analytics / Data scientist intern (work f...   
8                        Data Scientist - IBM Garage   
9      Digital Enterprise Architect (Data Scientist)   

                         company_name  \
0  Inflexion Analytix Private Limited   
1                              RedBus   
2                              RedBus   
3                   Fractal Analytics   
4   Morgan Stanley Advantage Services   
5                                FICO   
6                   Fractal Analytics   
7                      TalkValley LLC   
8              IBM India Pvt. Limited   
9                     Mphasis Limited   

                                            location  \
0  Mumbai, Hyderabad/Secunderabad, Pune, Gurgaon/...   
1                                Bangalore/Bengaluru   
2                                Bangalore/Bengaluru   
3      Mumbai, Gurgaon/Gurugram, Bangalore/Bengaluru   
4                                Bangalore/Bengaluru   
5                                Bangalore/Bengaluru   
6  Gurgaon/Gurugram, Bangalore/Bengaluru, Mumbai ...   
7          Kolkata, Bangalore/Bengaluru, Delhi / NCR   
8                                Bengaluru/Bangalore   
9                                Bangalore/Bengaluru   

                                            job_desc  
0  Job description\nJob Role : Data Scientist/Dat...  
1                                      Not Available  
2                                      Not Available  
3  Job description\nServe as primary technical le...  
4  Job description\nAbout Us\nMorgan Stanley is a...  
5  Job description\nJob Summary\nThe candidate sh...  
6  Job description\nPosition Description:\nThe Ar...  
7  Job description\nWe are a group of tenured pro...  
8                                      Not Available  
9                                      Not Available

So here we can see that we have created the dataset for Data scientist jobs named job1.

In [29]:
driver.close()

# Ques 3

Q3: In this question you have to scrape data using the filters available on the 
webpage as shown below:

You have to use the location and salary filter.

You have to scrape data for “Data Scientist” designation for first 10 job results.

You have to scrape the job-title, job-location, company_name,
experience_required.

The location filter to be used is “Delhi/NCR”

The salary filter to be used is “3-6” lakhs

In [3]:
# let's first connect to the web driver
driver = webdriver.Chrome(r"C:\Users\Neha\Downloads\chromedriver_win32\chromedriver.exe")

In [4]:
url = "https://www.naukri.com"
driver.get(url)

In [5]:
# finding element for job search bar
search_job = driver.find_element_by_xpath("//input[@class='sugInp']")
search_job

<selenium.webdriver.remote.webelement.WebElement (session="c2c368b4da34894c5eb8bec1c4ef6f87", element="a8779365-0a35-4da6-847e-407cd43ea5a4")>

In [6]:
# write on search bar
search_job.send_keys('Data Scientist')

In [7]:
search_btn= driver .find_element_by_xpath("//button[@class='btn']")
search_btn

<selenium.webdriver.remote.webelement.WebElement (session="c2c368b4da34894c5eb8bec1c4ef6f87", element="0a982439-770e-4d25-81b7-83a17cfd4a3a")>

In [8]:
# clicking  using xpath function
search_btn=driver.find_element_by_class_name('btn')
search_btn.click()

In [10]:
#so let's extract all the tags having the job titles
title_t1=driver.find_elements_by_xpath("//a[@class='title fw500 ellipsis']")
title_t1

[<selenium.webdriver.remote.webelement.WebElement (session="c2c368b4da34894c5eb8bec1c4ef6f87", element="751e5e4d-7086-4ea2-a7d6-c017c8cbe787")>,
 <selenium.webdriver.remote.webelement.WebElement (session="c2c368b4da34894c5eb8bec1c4ef6f87", element="3211d460-ad4a-415c-91b2-46b22582cac4")>,
 <selenium.webdriver.remote.webelement.WebElement (session="c2c368b4da34894c5eb8bec1c4ef6f87", element="748419fb-caf1-4361-92ee-2661719e37f3")>,
 <selenium.webdriver.remote.webelement.WebElement (session="c2c368b4da34894c5eb8bec1c4ef6f87", element="e5e5d2ed-043e-4f05-8753-097c1b6c3bac")>,
 <selenium.webdriver.remote.webelement.WebElement (session="c2c368b4da34894c5eb8bec1c4ef6f87", element="b78f03e7-7df6-4f09-b2f6-0d173f7da2e8")>,
 <selenium.webdriver.remote.webelement.WebElement (session="c2c368b4da34894c5eb8bec1c4ef6f87", element="5913bb3e-25d2-403a-afe2-3543db56dd17")>,
 <selenium.webdriver.remote.webelement.WebElement (session="c2c368b4da34894c5eb8bec1c4ef6f87", element="e13fe8b7-5c5c-428b-852d-99

In [11]:
# extract the text of the job title  from the tags
job_titles=[]
for i in title_t1:
    if i.text is None:
        job_titles.append('Not')
    else:
        job_titles.append(i.text)
job_titles[:10]        

['Data analytics / Data scientist intern (work from Home)',
 'Senior Data Scientist / Lead (Predictive Analytics)',
 'Data Science- Associate Data Scientist',
 'Associate Data Scientist',
 'Senior Data Scientist',
 'Immediate job opening - Senior Data Scientist',
 'Senior Data Scientist',
 'Lead Data Scientist',
 'Lead Data Scientist',
 'Chaayos is Looking For Data Scientist']

In [12]:
# lets extract all the tags having company names
company_t1=driver.find_elements_by_xpath("//a[@class='subTitle ellipsis fleft']")
company_t1

[<selenium.webdriver.remote.webelement.WebElement (session="c2c368b4da34894c5eb8bec1c4ef6f87", element="1bcfa59f-8e5d-4e52-bc08-76b765f02f14")>,
 <selenium.webdriver.remote.webelement.WebElement (session="c2c368b4da34894c5eb8bec1c4ef6f87", element="723176ef-9ee5-4eb4-9572-060b5bbca96a")>,
 <selenium.webdriver.remote.webelement.WebElement (session="c2c368b4da34894c5eb8bec1c4ef6f87", element="7ff2b9b2-a633-4fa1-b836-d0a3ca3f0b7f")>,
 <selenium.webdriver.remote.webelement.WebElement (session="c2c368b4da34894c5eb8bec1c4ef6f87", element="8ae150df-c8c6-4ac5-b11b-3495c0fe63ff")>,
 <selenium.webdriver.remote.webelement.WebElement (session="c2c368b4da34894c5eb8bec1c4ef6f87", element="a8d927f6-7035-4bb0-bdd9-f6406e0f36d4")>,
 <selenium.webdriver.remote.webelement.WebElement (session="c2c368b4da34894c5eb8bec1c4ef6f87", element="41459b73-d347-421d-baf1-cfe618c64ec3")>,
 <selenium.webdriver.remote.webelement.WebElement (session="c2c368b4da34894c5eb8bec1c4ef6f87", element="ff393844-7bff-4576-85db-c1

In [13]:
# Now we will extract the text from the tags by looping over these tags
companies_names=[]

for i in company_t1:
    companies_names.append(i.text)
companies_names[:10]    

['TalkValley LLC',
 'AVE-Promagne',
 'Jet2 Travel Technologies Pvt. Ltd.',
 'AlgoAnalytics Private Ltd.',
 'RedBus',
 'RedBus',
 'Fractal Analytics',
 'FICO',
 'Fractal Analytics',
 'Chaayos (Sunshine Teahouse Pvt. Ltd.)']

In [14]:
# so lets extract  all the tags having the experience required data
experience_t1=driver.find_elements_by_xpath("//li[@class='fleft grey-text br2 placeHolderLi experience'] //span")
experience_t1

[<selenium.webdriver.remote.webelement.WebElement (session="c2c368b4da34894c5eb8bec1c4ef6f87", element="dc8e4e5d-56af-46a3-bc1a-3b083833b76c")>,
 <selenium.webdriver.remote.webelement.WebElement (session="c2c368b4da34894c5eb8bec1c4ef6f87", element="ebd30d72-ef7e-4bfe-995c-2fb1b904cbb2")>,
 <selenium.webdriver.remote.webelement.WebElement (session="c2c368b4da34894c5eb8bec1c4ef6f87", element="99e7a9d2-af2a-4f80-a50d-e675444a91ea")>,
 <selenium.webdriver.remote.webelement.WebElement (session="c2c368b4da34894c5eb8bec1c4ef6f87", element="ab8a92c9-7996-4383-b5b4-d58a37d8e7c2")>,
 <selenium.webdriver.remote.webelement.WebElement (session="c2c368b4da34894c5eb8bec1c4ef6f87", element="5c28b3e6-9caf-4da6-a06c-82319a157790")>,
 <selenium.webdriver.remote.webelement.WebElement (session="c2c368b4da34894c5eb8bec1c4ef6f87", element="e0f44807-3b12-4e02-bd47-60f4c270188b")>,
 <selenium.webdriver.remote.webelement.WebElement (session="c2c368b4da34894c5eb8bec1c4ef6f87", element="174dd762-7ad3-431f-a051-67

In [15]:
# no we will extract the text  from these tags only by one by looping over these tags
experience_list=[]
for i in experience_t1:
    experience_list.append(i.text)
experience_list[:10]   

['0-5 Yrs',
 '7-10 Yrs',
 '0-2 Yrs',
 '0-1 Yrs',
 '5-8 Yrs',
 '5-8 Yrs',
 '5-9 Yrs',
 '6-11 Yrs',
 '10-15 Yrs',
 '0-5 Yrs']

In [17]:
# So lets extract all the tags having locations
locations_t1=driver.find_elements_by_xpath("//li[@class='fleft grey-text br2 placeHolderLi location']/span")
locations_t1

[<selenium.webdriver.remote.webelement.WebElement (session="c2c368b4da34894c5eb8bec1c4ef6f87", element="e4934abc-017f-4893-8113-e1e02428d2ea")>,
 <selenium.webdriver.remote.webelement.WebElement (session="c2c368b4da34894c5eb8bec1c4ef6f87", element="89040e51-c3eb-4936-82b4-cc808b8ea6f6")>,
 <selenium.webdriver.remote.webelement.WebElement (session="c2c368b4da34894c5eb8bec1c4ef6f87", element="a988a0a0-b0f8-4783-b1fa-c29caae626f9")>,
 <selenium.webdriver.remote.webelement.WebElement (session="c2c368b4da34894c5eb8bec1c4ef6f87", element="318b4f70-c8e0-4320-ba71-ce6796ee537a")>,
 <selenium.webdriver.remote.webelement.WebElement (session="c2c368b4da34894c5eb8bec1c4ef6f87", element="5b0b0916-de6c-40a2-90f0-a5f34be37f97")>,
 <selenium.webdriver.remote.webelement.WebElement (session="c2c368b4da34894c5eb8bec1c4ef6f87", element="65bd8ed9-5443-4bb3-a4c3-28a28afa679d")>,
 <selenium.webdriver.remote.webelement.WebElement (session="c2c368b4da34894c5eb8bec1c4ef6f87", element="cf5fa421-4e82-44be-9c07-44

In [18]:
#Now we wil extract the text from these tags only by one by looping over these tags
locations_list=[]
for i in locations_t1:
    locations_list.append(i.text)
locations_list[:10]    

['Kolkata, Bangalore/Bengaluru, Delhi / NCR',
 'Chennai',
 'Pune',
 'Pune(Baner)',
 'Bangalore/Bengaluru',
 'Bangalore/Bengaluru',
 'Mumbai, Gurgaon/Gurugram, Bangalore/Bengaluru',
 'Bangalore/Bengaluru',
 'Gurgaon/Gurugram, Bangalore/Bengaluru, Mumbai (All Areas)',
 'New Delhi']

In [19]:
#So lets check th length of ech element.
print(len(job_titles[:10])),print(len(companies_names[:10])),print(len(experience_list[:10])),print(len(locations_list[:10]))

10
10
10
10


(None, None, None, None)

# Creating a DataFarme for the Data Analyst jobs

In [20]:
jobs2=pd.DataFrame({})
jobs2['title']=job_titles[:10]
jobs2['company']=companies_names[:10]
jobs2['experience_required']=experience_list[:10]
jobs2['location']=locations_list[:10]

In [21]:
jobs2

title  \
0  Data analytics / Data scientist intern (work f...   
1  Senior Data Scientist / Lead (Predictive Analy...   
2             Data Science- Associate Data Scientist   
3                           Associate Data Scientist   
4                              Senior Data Scientist   
5      Immediate job opening - Senior Data Scientist   
6                              Senior Data Scientist   
7                                Lead Data Scientist   
8                                Lead Data Scientist   
9              Chaayos is Looking For Data Scientist   

                                 company experience_required  \
0                         TalkValley LLC             0-5 Yrs   
1                           AVE-Promagne            7-10 Yrs   
2     Jet2 Travel Technologies Pvt. Ltd.             0-2 Yrs   
3             AlgoAnalytics Private Ltd.             0-1 Yrs   
4                                 RedBus             5-8 Yrs   
5                                 RedBus             5-8 Yrs   
6                      Fractal Analytics             5-9 Yrs   
7                                   FICO            6-11 Yrs   
8                      Fractal Analytics           10-15 Yrs   
9  Chaayos (Sunshine Teahouse Pvt. Ltd.)             0-5 Yrs   

                                            location  
0          Kolkata, Bangalore/Bengaluru, Delhi / NCR  
1                                            Chennai  
2                                               Pune  
3                                        Pune(Baner)  
4                                Bangalore/Bengaluru  
5                                Bangalore/Bengaluru  
6      Mumbai, Gurgaon/Gurugram, Bangalore/Bengaluru  
7                                Bangalore/Bengaluru  
8  Gurgaon/Gurugram, Bangalore/Bengaluru, Mumbai ...  
9                                          New Delhi

We can see that we have created the dataset named jobs2.

In [22]:
driver.close()

# Ques4

Q4: Write a python program to scrape data for first 10 job results for Data scientist 
Designation in Noida location. You have to scrape company_name, No. of days 
ago when job was posted, Rating of the company.

In [9]:
import requests

In [10]:
page=requests.get("https://www.glassdoor.co.in/index.htm")
page

<Response [403]>

we are getting response 403 . so we can't scrape the data from this website.

# Ques5

In [11]:
pag1 =requests.get("https://www.glassdoor.co.in/Salaries/index.htm")
pag1

<Response [403]>

here also we are getting respponse 403.

# Ques 6

Q6 : Scrape data of first 100 sunglasses listings on flipkart.com. You have to 
scrape four attributes:
    
1. Brand

2. Product Description

3. Price

4. Discount %

The attributes which you have to scrape is ticked marked in the below image.

In [12]:
# let's first connect to the web driver
driver = webdriver.Chrome(r"C:\Users\Neha\Downloads\chromedriver_win32\chromedriver.exe")

In [13]:
url="https://www.flipkart.com/"
driver.get(url)

In [14]:
# finding element for job search bar
search_g= driver.find_element_by_xpath("//input[@type='text']")
search_g

<selenium.webdriver.remote.webelement.WebElement (session="7de1388e5d05f853d22155b31da7a0dc", element="4c50b02a-7a13-41a4-ab43-5783dea9b58e")>

In [15]:
# write on search bar
search_g.send_keys('sunglasses')

In [16]:
search_btn=driver.find_element_by_xpath("//button[@class='L0Z3Pu']")
search_btn

<selenium.webdriver.remote.webelement.WebElement (session="7de1388e5d05f853d22155b31da7a0dc", element="6ff1a9a5-5dcd-4dce-959c-3a7ae8563e95")>

In [17]:
search_btn=driver.find_element_by_class_name('L0Z3Pu')
search_btn.click()

In [23]:
B_name=[]
Price=[]
P_desc=[]
Discount=[]

In [25]:
for i in range(3):
    b_name=driver.find_elements_by_xpath("//div[@class='_2WkVRV']")
    p_desc=driver.find_elements_by_xpath("//a[@class='IRpwTa']")
    price =driver.find_elements_by_xpath("//div[@class='_25b18c']")
    discount=driver.find_elements_by_xpath("//div[@class='_3Ay6Sb']")
    
    for j  in b_name:
        B_name.append(j.text)
    B_name[:100]    
    
    
    
    for k in p_desc:
        P_desc.append(k.text)
    P_desc[:100] 
    
    
    for l in price:
        Price.append(l.text)
    Price[:100] 
    
    
    for t in discount:
        Discount.append(t.text)
    Discount[:100]

In [26]:
B_name[:100]
    

['PIRASO',
 'Fastrack',
 'Fastrack',
 'kingsunglasses',
 'PIRASO',
 'PIRASO',
 'Fastrack',
 'PIRASO',
 'DEIXELS',
 'GANSTA',
 'ROZZETTA CRAFT',
 'Fastrack',
 'PIRASO',
 'Singco',
 'elegante',
 'NuVew',
 'Silver Kartz',
 'ROZZETTA CRAFT',
 'PHENOMENAL',
 'PHENOMENAL',
 'HIPPON',
 'Singco India',
 'Singco India',
 'ROYAL SON',
 'ROYAL SON',
 'Singco India',
 'ROZZETTA CRAFT',
 'NuVew',
 'Fastrack',
 'Singco India',
 'ROYAL SON',
 'hipe',
 'ROZZETTA CRAFT',
 'ROYAL SON',
 'NuVew',
 'GANSTA',
 'PHENOMENAL',
 'ROZZETTA CRAFT',
 'ROZZETTA CRAFT',
 'PHENOMENAL',
 'PIRASO',
 'Fastrack',
 'Fastrack',
 'kingsunglasses',
 'PIRASO',
 'PIRASO',
 'Fastrack',
 'PIRASO',
 'DEIXELS',
 'GANSTA',
 'ROZZETTA CRAFT',
 'Fastrack',
 'PIRASO',
 'Singco',
 'elegante',
 'NuVew',
 'Silver Kartz',
 'ROZZETTA CRAFT',
 'PHENOMENAL',
 'PHENOMENAL',
 'HIPPON',
 'Singco India',
 'Singco India',
 'ROYAL SON',
 'ROYAL SON',
 'Singco India',
 'ROZZETTA CRAFT',
 'NuVew',
 'Fastrack',
 'Singco India',
 'ROYAL SON',
 'hipe'

In [27]:
print(len(B_name[:100])),print(len(Price[:100])),print(len(P_desc[:100])),print(len(Discount[:100]))

100
100
100
100


(None, None, None, None)

# Creating a dataframe of the above data

In [29]:
sun_gl=pd.DataFrame({})
sun_gl['Brand_name']=B_name[:100]
sun_gl['P_price']=Price[:100]
sun_gl['Pr_desc']=P_desc[:100]
sun_gl['P_discount']=Discount[:100]

In [31]:
sun_gl

Brand_name            P_price  \
0           PIRASO  ₹237₹1,59985% off   
1         Fastrack    ₹570₹79928% off   
2         Fastrack    ₹621₹79922% off   
3   kingsunglasses  ₹299₹2,59988% off   
4           PIRASO  ₹237₹1,59985% off   
..             ...                ...   
95           NuVew  ₹331₹1,59979% off   
96    Silver Kartz    ₹733₹89918% off   
97  ROZZETTA CRAFT  ₹331₹2,59987% off   
98      PHENOMENAL  ₹213₹1,19982% off   
99      PHENOMENAL  ₹281₹1,99985% off   

                                              Pr_desc P_discount  
0               UV Protection Aviator Sunglasses (54)    85% off  
1    UV Protection Rectangular Sunglasses (Free Size)    28% off  
2   Gradient, UV Protection Wayfarer Sunglasses (F...    22% off  
3   Mirrored, UV Protection Wayfarer Sunglasses (F...    88% off  
4               UV Protection Aviator Sunglasses (54)    85% off  
..                                                ...        ...  
95              UV Protection Aviator Sunglasses (57)    79% off  
96      UV Protection Wayfarer Sunglasses (Free Size)    83% off  
97  UV Protection, Gradient Rectangular Sunglasses...    77% off  
98  UV Protection, Mirrored Retro Square Sunglasse...    80% off  
99  UV Protection Retro Square Sunglasses (Free Size)    80% off  

[100 rows x 4 columns]

#We can see that we have created the dataframe.

In [32]:
driver.close()

# Q8: Scrape data for first 100 sneakers you find when you visit flipkart.com and 
search for “sneakers” in the search field.

You have to scrape 4 attributes of each sneaker :
    
1. Brand

2. Product Description

3. Price

4. discount %

In [3]:
# let's first connect to the web driver
driver = webdriver.Chrome(r"C:\Users\Neha\Downloads\chromedriver_win32\chromedriver.exe")

In [4]:
url="https://www.flipkart.com/"
driver.get(url)

In [5]:
# finding element for job search bar
search_g= driver.find_element_by_xpath("//input[@type='text']")
search_g

<selenium.webdriver.remote.webelement.WebElement (session="0003341c3c278ca61eb98703d1675fbf", element="55e14e1c-09fa-4154-96b0-caf0cf0b4c4e")>

In [6]:
# write on search bar
search_g.send_keys('sneakers')

In [7]:
search_btn=driver.find_element_by_xpath("//button[@class='L0Z3Pu']")
search_btn

<selenium.webdriver.remote.webelement.WebElement (session="0003341c3c278ca61eb98703d1675fbf", element="827dc307-afd7-4cc9-97a1-7cb0f92e6a55")>

In [8]:
search_btn=driver.find_element_by_class_name('L0Z3Pu')
search_btn.click()

In [9]:
B_name=[]
Price=[]
P_desc=[]
Discount=[]

In [10]:
for i in range(3):
    b_name=driver.find_elements_by_xpath("//div[@class='_2WkVRV']")
    p_desc=driver.find_elements_by_xpath("//a[@class='IRpwTa']")
    price =driver.find_elements_by_xpath("//div[@class='_25b18c']")
    discount=driver.find_elements_by_xpath("//div[@class='_3Ay6Sb']")
    
    for j  in b_name:
        B_name.append(j.text)
    B_name[:100]    
    
    
    
    for k in p_desc:
        P_desc.append(k.text)
    P_desc[:100] 
    
    
    for l in price:
        Price.append(l.text)
    Price[:100] 
    
    
    for t in discount:
        Discount.append(t.text)
    Discount[:100]

In [11]:
print(len(B_name[:100])),print(len(Price[:100])),print(len(P_desc[:100])),print(len(Discount[:100]))

100
100
100
100


(None, None, None, None)

# Creating a dataframe of the above data

In [13]:
sun_gl=pd.DataFrame({})
sun_gl['Brand_name']=B_name[:100]
sun_gl['P_price']=Price[:100]
sun_gl['Pr_desc']=P_desc[:100]
sun_gl['P_discount']=Discount[:100]

In [14]:
sun_gl

Brand_name            P_price  \
0        Chevit  ₹448₹1,49870% off   
1     India hub  ₹389₹2,99987% off   
2      CALCADOS  ₹748₹1,99662% off   
3        ORICUM    ₹377₹99862% off   
4    Shoes Bank    ₹349₹99965% off   
..          ...                ...   
95        3SIX5    ₹474₹99952% off   
96        Birde    ₹399₹99960% off   
97   restinfoot    ₹499₹99950% off   
98     ASTEROID    ₹377₹99962% off   
99  kardam&sons    ₹242₹69965% off   

                                              Pr_desc P_discount  
0   Unique & Perfect Collection Combo Pack of 02 S...    70% off  
1   Combo pack of 2 casual sneaker shoes for men S...    87% off  
2      White Sneaker For Men's/Boy's Sneakers For Men    62% off  
3                                    Sneakers For Men    62% off  
4                                    Sneakers For Men    65% off  
..                                                ...        ...  
95  Casual , Partywear Sneakers Shoes For Men's An...    57% off  
96         casual for men (beige 06) Sneakers For Men    43% off  
97                                   Sneakers For Men    62% off  
98  Combo pack of 2 Casuals Shoes For Men (1563-15...    75% off  
99  Combo Pack of 4 Latest Collection Stylish Casu...    81% off  

[100 rows x 4 columns]

We have created the dataframe

In [15]:
driver.close()

# Q9: Go to the link - https://www.myntra.com/shoes
Set Price filter to “Rs. 6649 to Rs. 13099” , Color filter to “Black”,

And then scrape First 100 shoes data you get. The data should include “Brand” of 
the shoes , Short Shoe description, price of the shoe 

In [3]:
# let's first connect to the web driver
driver = webdriver.Chrome(r"C:\Users\Neha\Downloads\chromedriver_win32\chromedriver.exe")

In [4]:
url="https://www.myntra.com/shoes"
driver.get(url)

In [8]:
# appplying the price filter
filter_button=driver.find_elements_by_xpath("//label[@class='common-customerCheckbox vertical-filters-label']")
for i in filter_button:
    if i.text=="Rs. 6649 to Rs. 13099":
        i.click()
        break

In [11]:
#Applying the black colur filter
filter_button=driver.find_elements_by_xpath("//li[@class='colour-listItem']")
for i in filter_button:
    if i.text=="Black":
        i.click()
        break

In [12]:
B_name=[]
Price=[]
P_desc=[]

In [29]:
for i in range(2):
    b_name=driver.find_elements_by_xpath("//h1[@class='pdp-title']")
    p_desc=driver.find_elements_by_xpath("//h1[@class='name']")
    price =driver.find_elements_by_xpath("//div[@class='product-price']")
 
    for j  in b_name:
        B_name.append(j.text)
    B_name[:100]    
    
    
    
    for k in p_desc:
        P_desc.append(k.text)
    P_desc[:100] 
    
    
    for l in price:
        Price.append(l.text)
    Price[:100] 
    

In [30]:
print(len(B_name[:100])),print(len(Price[:100])),print(len(P_desc[:100]))

2
100
0


(None, None, None)

# Q10: Go to webpage https://www.amazon.in/
 Enter “Laptop” in the search field and then click the search icon.
 Then set CPU Type filter to “Intel Core i7” and “Intel Core i9” as

In [3]:
# let's first connect to the web driver
driver = webdriver.Chrome(r"C:\Users\Neha\Downloads\chromedriver_win32\chromedriver.exe")

In [4]:
url=" https://www.amazon.in "
driver.get(url)

In [6]:
# finding element for job search bar
search_g= driver.find_element_by_xpath("//input[@type='text']")
search_g

<selenium.webdriver.remote.webelement.WebElement (session="a2ad6e875a1461b017453195fbca0aea", element="3e0f7969-cd34-4a29-9d89-4c7f9e081e64")>

In [7]:
# write on search bar
search_g.send_keys('Laptop')

In [8]:
search_btn=driver.find_element_by_xpath("//input[@id='nav-search-submit-button']")
search_btn

<selenium.webdriver.remote.webelement.WebElement (session="a2ad6e875a1461b017453195fbca0aea", element="c44804a4-6a54-4edf-947e-0d869dc08667")>

In [9]:
search_btn=driver.find_element_by_xpath("//input[@id='nav-search-submit-button']")
search_btn.click()

No core filter are shown on the website

In [ ]:
Title=[]
Price=[]
Rating=[]

In [ ]:
for i in range(3):
    b_name=driver.find_elements_by_xpath("//div[@class='_2WkVRV']")
    p_desc=driver.find_elements_by_xpath("//a[@class='IRpwTa']")
    price =driver.find_elements_by_xpath("//div[@class='_25b18c']")
    
    
    for j  in b_name:
        Title.append(j.text)
    Title[:100]    
    
    
    
    for k in p_desc:
        P_desc.append(k.text)
    P_desc[:100] 
    
    
    for l in price:
        Price.append(l.text)
    Price[:100] 
    